app/api/saml/metadata/route.ts

```diff
@@ -0,0 +1,20 @@
+import { NextRequest, NextResponse } from 'next/server'
+import { saml } from '@/lib/saml-config'
+
+export async function GET(request: NextRequest) {
+  try {
+    const metadata = saml.generateServiceProviderMetadata(
+      process.env.SAML_SP_CERT || null,
+      process.env.SAML_SP_CERT || null
```

Comment:
You don't really want to generate metadata without the certs do you? Perhaps some validation and error handling.